integrationtests/agent/watcher_trigger_test.go

```diff
@@ -0,0 +1,194 @@
+			Expect(env.k8sClient.Update(ctx, cm)).ToNot(HaveOccurred())
+			Eventually(func() int {
+				return *triggerCount
+			}).WithPolling(100 * time.Millisecond).MustPassRepeatedly(3).
+				Should(Equal(1))
+
+			cm.Data = map[string]string{"bar": "baz"}
+			Expect(env.k8sClient.Update(ctx, cm)).ToNot(HaveOccurred())
+			Eventually(func() int {
+				return *triggerCount
+			}).WithPolling(100 * time.Millisecond).MustPassRepeatedly(3).
+				Should(Equal(2))
+		})
+	})
+	When("watching a deployed deployment", Ordered, func() {
+		createDeployment := func() *appsv1.Deployment {
+			deploy := appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "test-deploy-",
+					Namespace:    env.namespace,
+					Generation:   1,
+				},
+				Spec: appsv1.DeploymentSpec{
+					Replicas: pointer.Int32(0),
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"app": "test",
+						},
+					},
+					Paused: true,
+					Template: corev1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								"app": "test",
+							},
+						},
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  "test",
+									Image: "test-image",
+								},
+							},
+						},
+					},
+				},
+			}
+
+			err := env.k8sClient.Create(ctx, &deploy)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(deploy.UID).ToNot(BeEmpty())
+			// envtest does not return a complete object, which is needed for the trigger to work
+			deploy.TypeMeta = metav1.TypeMeta{
+				Kind:       "Deployment",
+				APIVersion: "apps/v1",
+			}
+			return &deploy
+		}
+		var triggerCount *int
+		var deploy *appsv1.Deployment
+
+		BeforeEach(func() {
+			deploy = createDeployment()
+			triggerCount = registerResource(env.namespace+"/test-deploy", deploy)
+			DeferCleanup(func() {
+				Expect(
+					client.IgnoreNotFound(env.k8sClient.Delete(ctx, deploy))).
+					NotTo(HaveOccurred())
+			})
+		})
+		It("is not initially triggered", func() {
+			Consistently(func() int {
+				return *triggerCount
+			}).Should(Equal(0))
+		})
+		It("is triggered on deletion", func() {
+			Expect(env.k8sClient.Delete(ctx, deploy)).ToNot(HaveOccurred())
+			Eventually(func() int {
+				return *triggerCount
+			}).WithPolling(100 * time.Millisecond).MustPassRepeatedly(3).
+				Should(Equal(1))
+		})
+		It("is not triggered on status updates", func() {
+			deploy.Status.Conditions = []appsv1.DeploymentCondition{
+				{
+					Type:               appsv1.DeploymentAvailable,
+					Status:             corev1.ConditionFalse,
+					LastUpdateTime:     metav1.Now(),
+					LastTransitionTime: metav1.Now(),
+					Message:            "tests",
+					Reason:             "tests",
+				},
+			}
+			Expect(env.k8sClient.Status().Update(ctx, deploy)).ToNot(HaveOccurred())
+			Consistently(func() int {
+				return *triggerCount
+			}).Should(Equal(0))
+		})
+		It("is triggered Spec is changed", func() {
```

Comment:
nit:
```suggestion
		It("is triggered when Spec is changed", func() {
```
or (more consistent with the previous test description):
```suggestion
		It("is triggered on Spec changes", func() {
```